controllers/preflightvalidationocp_reconciler.go

```diff
@@ -0,0 +1,271 @@
+	if err != nil {
+		log.Error(err, "runPreflightValidationOCP failed")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *PreflightValidationOCPReconciler) runPreflightValidationOCP(ctx context.Context, pvo *kmmv1beta1.PreflightValidationOCP) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	// get compatible PreflightValidation
+	nsn := types.NamespacedName{Name: pvo.Name, Namespace: pvo.Namespace}
+	pv := &kmmv1beta1.PreflightValidation{}
+	err := r.client.Get(ctx, nsn, pv)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Compatible PreflightValidation not found, creating")
+			pv, err = r.preparePreflightValidation(ctx, pvo)
+			if err != nil {
+				return fmt.Errorf("failed to prepare the data for compatible PreflightValidation: %v", err)
+			}
+			err = r.client.Create(ctx, pv)
+			if err != nil {
+				return fmt.Errorf("failed to create PreflightValidation %s with kernel version %s: %v", nsn, pv.Spec.KernelVersion, err)
+			}
+			return nil
+		}
+		return fmt.Errorf("failed to get the compatible PreflightValidation, %s: %v", nsn, err)
+	}
+
+	log.Info("Updating status from PV ", "pv", nsn, "pvo status", pvo.Status)
+
+	// update statuses
+	err = r.statusUpdater.PreflightOCPUpdateStatus(ctx, pvo, pv)
+	if err != nil {
+		return fmt.Errorf("failed to update the statuses from compatible PreflightValidations: %v", err)
+	}
+
+	// no need to check all the modules verification statuses, it will be done by preflight validation reconciler
+	return nil
+}
+
+func (r *PreflightValidationOCPReconciler) preparePreflightValidation(ctx context.Context,
+	pvo *kmmv1beta1.PreflightValidationOCP) (*kmmv1beta1.PreflightValidation, error) {
+	log := ctrl.LoggerFrom(ctx)
+	dtkImage, err := r.getDTKFromImage(ctx, pvo.Spec.ReleaseImage)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get DTK image from Release Image %s: %v", pvo.Spec.ReleaseImage, err)
+	}
+
+	log.Info("DTK image is", "dtk_image", dtkImage)
+
+	kernelVersion, osVersion, err := r.getKernelVersionAndOSFromDTK(ctx, dtkImage)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get kernel/os version from DTK image %s: %v", dtkImage, err)
+	}
+
+	log.Info("os version and kernel version", "os_version", osVersion, "kernel_version", kernelVersion)
+
+	// os version in the DTK is different then OS version on the node, so in case regular flow
+	// already set the data - we don't want to override it
+	if _, err := r.kernelOsDtkMapping.GetImage(kernelVersion); err != nil {
+		r.kernelOsDtkMapping.SetNodeInfo(kernelVersion, osVersion)
+		r.kernelOsDtkMapping.SetImageStreamInfo(osVersion, dtkImage)
+	}
+
+	pv := kmmv1beta1.PreflightValidation{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pvo.Name,
+			Namespace: pvo.Namespace,
+		},
+		Spec: kmmv1beta1.PreflightValidationSpec{
+			KernelVersion:  kernelVersion,
+			PushBuiltImage: pvo.Spec.PushBuiltImage,
+		},
+	}
+
+	err = controllerutil.SetControllerReference(pvo, &pv, r.scheme)
```

Comment:
```suggestion
	if err := controllerutil.SetControllerReference(pvo, &pv, r.scheme); err != nil {
```

Reply:
I also prefer that style.

Reply:
I understand that several `PreflightValidationOCP` objects can own a `PreflightValidation`? If so, I think we should use [`SetOwnerReference`](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil#SetOwnerReference) instead.

Reply:
I prefer to set the err != nil on a different line, i think it is a little bit more readable

Reply:
@qbarrand one PreflightValidationOCP owns exactly one Preflight